examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,574 @@
+```
+# The dataprep service will add a .txt postfix for link file
+
+curl -X POST "http://localhost:6007/v1/dataprep/delete_file" \
+     -d '{"file_path": "https://opea.dev.txt"}' \
+     -H "Content-Type: application/json"
+```
+
+#### Delete file
+
+```
+curl -X POST "http://localhost:6007/v1/dataprep/delete_file" \
+     -d '{"file_path": "nke-10k-2023.pdf"}' \
+     -H "Content-Type: application/json"
+```
+
+#### Delete all uploaded files and links
+
+```
+curl -X POST "http://localhost:6007/v1/dataprep/delete_file" \
+     -d '{"file_path": "all"}' \
+     -H "Content-Type: application/json"
+```
+
+### TEI Embedding Service
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-tei 6006:80
+```
+Follow the below steps in a different terminal.
+
+The TEI embedding service takes in a string as input, embeds the string into a
+vector of a specific length determined by the embedding model and returns this
+embedded vector.
+
+```
+curl http://localhost:6006/embed \
+    -X POST \
+    -d '{"inputs":"What is Deep Learning?"}' \
+    -H 'Content-Type: application/json'
+```
+
+In this example the embedding model used is "BAAI/bge-base-en-v1.5", which has a vector size of 768. So the output of the curl command is a embedded vector of
+length 768.
+
+### Embedding Microservice
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-embedding-usvc 6000:6000
+```
+Follow the below steps in a different terminal.
+
+The embedding microservice depends on the TEI embedding service. In terms of
+input parameters, it takes in a string, embeds it into a vector using the TEI
+embedding service and pads other default parameters that are required for the
+retrieval microservice and returns it.
+```
+curl http://localhost:6000/v1/embeddings\
+  -X POST \
+  -d '{"text":"hello"}' \
+  -H 'Content-Type: application/json'
+```
+
+### Retriever Microservice
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-retriever-usvc 7000:7000
+```
+Follow the below steps in a different terminal.
+
+To consume the retriever microservice, you need to generate a mock embedding
+vector by Python script. The length of embedding vector is determined by the
+embedding model. Here we use the
+model EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5", which vector size is 768.
+
+Check the vector dimension of your embedding model and set
+`your_embedding` dimension equal to it.
+
+```
+export your_embedding=$(python3 -c "import random; embedding = [random.uniform(-1, 1) for _ in range(768)]; print(embedding)")
+
+curl http://localhost:7000/v1/retrieval \
+  -X POST \
+  -d "{\"text\":\"test\",\"embedding\":${your_embedding}}" \
+  -H 'Content-Type: application/json'
```

Comment:
You could probably add the use of `jq` here to truncate just the text output. So it's more readable.